Common/src/linear_solvers_structure.cpp

```diff
@@ -651,6 +652,95 @@ unsigned long CSysSolve<ScalarType>::BCGSTAB_LinSolver(const CSysVector<ScalarTy
   return (unsigned long) i;
 }
 
+template<class ScalarType>
+unsigned long CSysSolve<ScalarType>::Smoother_LinSolver(const CSysVector<ScalarType> & b, CSysVector<ScalarType> & x,
+                                                        CMatrixVectorProduct<ScalarType> & mat_vec, CPreconditioner<ScalarType> & precond,
+                                                        ScalarType tol, unsigned long m, ScalarType *residual, bool monitoring, CConfig *config) {
+
+  int rank = SU2_MPI::GetRank();
+  ScalarType norm_r = 0.0, norm0 = 0.0, omega = 1.0; // hardcoded relaxation factor
```

Comment:
Have you investigated the effects of the relaxation factor? When I first put this in, I included a variable for it, but I am not sure I tested it much. Would be interested to see.

Also might be worth exposing this to the config and/or making it consistent with the relaxation factors appearing in other linear solvers.

Reply:
Not in SU2, but the theory should be that if the preconditioner does not approximate the inverse of the matrix well you need smaller relaxation, you would want the relaxation that minimizes the spectral radius but that is not easy to compute. I also do not know what is the effect on the "quality" of the smoothing...

I can expose it to the config but I'm not too keen on running parameters tests on this.
By consistent with the other solvers do you mean computing it every iteration with some formula (e.g. such that the norm of the residual is minimized)? With this I have played with and it very easily stalls (omega=0).

Thank you for starting the review this quickly by the way.

Reply:
Aok, no need to test rigorously. I would be interested to have this exposed to the config though.

As for consistency with other relaxation factors, it is probably beyond this PR, and it can wait for later. Basically, what I meant is that it would be nice to see a more thought-out strategy for relaxation factors that affect the linear solvers. The current approach seems disjointed, as we have RELAXATION_FACTOR_FLOW that multiplies the output increment of the linear solve in the FV flow solver, for instance, and then other potential relaxation factors inside the linear solver, such as the one in question, and maybe other physics solvers that don't apply relaxation at all in the linear solver. Just food for thought for now.